pika/connection.py

```diff
@@ -1783,52 +1820,50 @@ def _send_frame(self, frame_value):
         if self.params.backpressure_detection:
             self._detect_backpressure()
 
-    def _send_method(self, channel_number, method_frame, content=None):
+    def _send_method(self, channel_number, method, content=None):
         """Constructs a RPC method frame and then sends it to the broker.
 
         :param int channel_number: The channel number for the frame
-        :param pika.object.Method method_frame: The method frame to send
+        :param pika.amqp_object.Method method: The method to send
         :param tuple content: If set, is a content frame, is tuple of
                               properties and body.
 
         """
-        if not content:
-            with self._write_lock:
-                self._send_frame(frame.Method(channel_number, method_frame))
-                return
-        self._send_message(channel_number, method_frame, content)
+        if content:
+            self._send_message(channel_number, method, content)
+        else:
+            self._send_frame(frame.Method(channel_number, method))
 
-    def _send_message(self, channel_number, method_frame, content=None):
+    def _send_message(self, channel_number, method, content=None):
         """Send the message directly, bypassing the single _send_frame
         invocation by directly appending to the output buffer and flushing
         within a lock.
 
         :param int channel_number: The channel number for the frame
-        :param pika.object.Method method_frame: The method frame to send
+        :param pika.amqp_object.Method method: The method frame to send
         :param tuple content: If set, is a content frame, is tuple of
                               properties and body.
 
         """
         length = len(content[1])
-        write_buffer = [frame.Method(channel_number, method_frame).marshal(),
+        write_buffer = [frame.Method(channel_number, method).marshal(),
                         frame.Header(channel_number, length,
                                      content[0]).marshal()]
         if content[1]:
             chunks = int(math.ceil(float(length) / self._body_max_length))
-            for chunk in range(0, chunks):
-                s = chunk * self._body_max_length
-                e = s + self._body_max_length
-                if e > length:
-                    e = length
+            for chunk in xrange(0, chunks):
+                start = chunk * self._body_max_length
+                end = start + self._body_max_length
+                if end > length:
+                    end = length
                 write_buffer.append(frame.Body(channel_number,
-                                               content[1][s:e]).marshal())
-
-        with self._write_lock:
```

Comment:
REVIEWME removed write-locking, since Connection is not thread-safe by design, and a number of other accesses to the output buffer weren't guarded anyway.